api/desecapi/views/domains.py

```diff
@@ -86,6 +101,12 @@ def perform_destroy(self, instance: Domain):
             with PDNSChangeTracker():
                 parent_domain.update_delegation(instance)
 
+    @action(detail=True, renderer_classes=[PlainTextRenderer])
+    def zonefile(self, request, name=None):
+        instance = self.get_object()
```

Comment:
self.domain?

Reply:
The canonical thing in DRF is actually `self.get_object()`. I added the `domain` property only because the `TokenHasDomainRRsetsPermission` permission expects the view to have it.